authority/authorize.go

```diff
@@ -91,123 +90,171 @@ func (a *Authority) authorizeToken(ctx context.Context, ott string) (provisioner
 	ctx := provisioner.NewContextWithMethod(context.Background(), provisioner.SignMethod)
-	return a.Authorize(ctx, ott)
+	return a.Authorize(ctx, token)
 }
 
-// authorizeRevoke authorizes a revocation request by validating and authenticating
-// the RevokeOptions POSTed with the request.
-// Returns a tuple of the provisioner ID and error, if one occurred.
+// authorizeRevoke locates the provisioner used to generate the authenticating
+// token and then performs the token validation flow.
 func (a *Authority) authorizeRevoke(ctx context.Context, token string) error {
-	errContext := map[string]interface{}{"ott": token}
+	var opts = []errs.Option{errs.WithKeyVal("token", token)}
 
 	p, err := a.authorizeToken(ctx, token)
 	if err != nil {
-		return &apiError{errors.Wrap(err, "authorizeRevoke"), http.StatusUnauthorized, errContext}
+		return errs.Wrap(http.StatusInternalServerError, err, "authority.authorizeRevoke", opts...)
 	}
 	if err = p.AuthorizeRevoke(ctx, token); err != nil {
-		return &apiError{errors.Wrap(err, "authorizeRevoke"), http.StatusUnauthorized, errContext}
+		return errs.Wrap(http.StatusInternalServerError, err, "authority.authorizeRevoke", opts...)
 	}
 	return nil
 }
 
-// authorizeRenewl tries to locate the step provisioner extension, and checks
+// authorizeRenew locates the provisioner (using the provisioner extension in the cert), and checks
 // if for the configured provisioner, the renewal is enabled or not. If the
 // extra extension cannot be found, authorize the renewal by default.
 //
 // TODO(mariano): should we authorize by default?
-func (a *Authority) authorizeRenew(crt *x509.Certificate) error {
-	errContext := map[string]interface{}{"serialNumber": crt.SerialNumber.String()}
+func (a *Authority) authorizeRenew(cert *x509.Certificate) error {
+	var opts = []errs.Option{errs.WithKeyVal("serialNumber", cert.SerialNumber.String())}
 
 	// Check the passive revocation table.
-	isRevoked, err := a.db.IsRevoked(crt.SerialNumber.String())
+	isRevoked, err := a.db.IsRevoked(cert.SerialNumber.String())
 	if err != nil {
-		return &apiError{
-			err:     errors.Wrap(err, "renew"),
-			code:    http.StatusInternalServerError,
-			context: errContext,
-		}
+		return errs.Wrap(http.StatusInternalServerError, err, "authority.authorizeRenew", opts...)
 	}
 	if isRevoked {
-		return &apiError{
-			err:     errors.New("renew: certificate has been revoked"),
-			code:    http.StatusUnauthorized,
-			context: errContext,
-		}
+		return errs.Unauthorized(errors.New("authority.authorizeRenew: certificate has been revoked"), opts...)
 	}
 
-	p, ok := a.provisioners.LoadByCertificate(crt)
+	p, ok := a.provisioners.LoadByCertificate(cert)
 	if !ok {
-		return &apiError{
-			err:     errors.New("renew: provisioner not found"),
-			code:    http.StatusUnauthorized,
-			context: errContext,
-		}
+		return errs.Unauthorized(errors.New("authority.authorizeRenew: provisioner not found"), opts...)
```

Comment:
Let me preface this by saying - you're right.

Specifically in this case, I would likely have had to do these commits sequentially (because there was a lot of inter related code) which probably would have taken longer. 

I'll do my best to keep logical commits separated in the future.